File: 프로그래머스/unrated/181880. 1로 만들기/1로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    int count = 0;
    
    for (int i = 0; i < num_list.size(); i++)
    {
        while(num_list[i] != 1)
        {
            if (num_list[i] % 2)
            {
                num_list[i] -= 1;
                num_list[i] /= 2;
            }
            else
                num_list[i] /= 2;
            count++;
        }
    }
    
    return count;
}

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        
        for (int i = 0; i < num_list.length; i++) {
            answer += go(num_list[i]);
        }
        
        return answer;
    }
    
    private int go(int n) {
        int count = 0;
        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2;  
            } else {
                n = (n - 1) / 2;
            }
            count++;
        }
        return count;
    }
}